### ISSUE TITLE
Collecting activity data for deeper analysis 

### ISSUE THREAD

**Issue by leoch20** [InstaPy/InstaPy#3272]
## Expected Behavior
I have seen a lot of discussion about what the best settings for InstaPy are and everyone has a different approach. At the end what every approach you chose you need a way to evaluate the performance of your bot. Right now the not only collect very high level activity data, which does not tell you if interacting with a given user or a hashtag is useful. 

I would like to have the bot collecting every "important" interaction it does. Even if it is hundreds per day. That way we could take a business intelligence approach to the bot's activity to better determine how we want it to behave.

## Current Behavior
InstaPy only collects high level logs.

## Possible Solution
Adding a table with just a few columns (These is what I came up with)
Columns: LogTime, Source, Action, User

`LogTime`  -> When the activity/log happened 
`Source `    -> Where the activity came from. Ex, interact with followers of Mike. Then Mike is the source
`Action `     ->  Liked, Commented, Followed, Unfollowed 
`User`         -> Who we interacted with. Ex, Mike's follower

## InstaPy configuration

This table is defined in `print_log_writer.py` and called in `InstaPy.py` and `unfollow_util.py:`

```
def log_table_activity(login, action, user, logger, logfolder, logtime, source):
    """Logs all interactions into
    a seperate file"""
    newFile = os.path.exists('{0}{1}_loggingActivity.csv'.format(logfolder, login)) == False
    try:
        with open('{0}{1}_loggingActivity.csv'.format(logfolder, login), 'a') as loggingActivityTable:
            with interruption_handler():
                if newFile:
                    loggingActivityTable.write('LogTime,Source,Action,User,\n')

                loggingActivityTable.write('{},{},{},{},\n'.format(logtime, source, action, user))
    except BaseException as e:
        logger.error("loggingActivityTable error {}".format(str(e)))

```

-----------
I have seen something like this requested before in [1480](https://github.com/timgrossmann/InstaPy/issues/1480) and [1417](https://github.com/timgrossmann/InstaPy/issues/1417).
My approach has been working for the past few days, but if anyone has something similar or better, I'm all ears.

Finally, in order to determine if whatever we are doing works we need to know if our goal was reached. For me it is if I got ppl to follow my account (I guess that's provably for most of us). In order to do that I am using [InstagramHelper](https://github.com/OllegK/InstagramHelper/blob/master/README.md) to get info about the accounts that follow me. 

I am populating a table and comparing it against my results from the new data collected from the bot. If anyone wants I can explain deeper how I am doing so.

Anyone interested in adding this to the bot?

**Comment by sionking** [comment]
I think some professional  users here do their analytics.
But you more then welcome to share.

**Comment by lucascaldax** [comment]
Hello there, I have been using a different approach. 
1-I collect data (in my case the followers of someone), only once. I research the best target based on the number of bots they have as followers, the number of daily posts and likes and the likeness as my insta.
2-After that, I split this into 200 names lists of people. Someone with 50k followers would generate 250 lists.
3-I follow 2 lists per day. So it would take 125 days to follow everyone.
4-3 days later, I unfollow the first 2 lists, only the nonfollowers.
5- After I reach something like 2.5k followers/2k following, I will start to unfollow even followers. So I can remain X followers/2k following forever.

Now I now that out of 200 people, only 14 to 23 give me a followback. But this means 28-46 per day, plus random followers, so it is not that bad. Next time I switch my target, I will be able to tell which one is the best target. I can also analyze which strategy is giving me more followers, Imagine if I do 10 batches with only likes and the next 10 with only comments. 

Two extra things, based on my impressions (I don't have numbers to really analyze it):
1-I stopped using likes in hashtags since it was attracting lots of bots, which would unfollow me later.
2-I stop skipping people/business/anything since it was costing me lots and lots of server calls, and my growth was penalized.  I imagine that this is like working out, I am in the phase of bulk, I don't care about the quality of my followers, as long as I have an engagement rate acceptable (mine is roughly 12 percent at the moment) which is pretty good.

Any thoughts will be appreciated.

**Comment by leoch20** [comment]
I added my changes so "all" the bot's activity is collected [here](https://github.com/timgrossmann/InstaPy/pull/3275)

@lucascaldax it sounds interesting. Can you elaborate a bit for your first step: 

> 1-I collect data (in my case the followers of someone), only once. I research the best target based on the number of bots they have as followers, the number of daily posts and likes and the likeness as my insta.

Are you evaluating what accounts you should interact with before you interact with them?
And, have you had any issues following and unfollowing 400 user per day?

**Comment by lucascaldax** [comment]
@leoch20 Nothing fancy. For example I am a photographer from Rio, so I chose a photographer from Rio with 50k followers. His photos r the same theme as mine. So I chose him. Its important to choose someone with a good engagement ratio..  look for someone between 1%-5%. 
Calculator:      https://phlanx.com/engagement-calculator
Auditing an account:    https://hypeauditor.com/
--> search for someone with authentic engagement in ur niche (Photography -Brazil e.g)
For example this dude i am targeting now, only gained 2k followers in the last year, so his hype is not that good. We should aim for at least 10k per year, without extra strategies).
You can definetly run surveys follows (50 follows in 8 possible targets) on some people and check which ones are the best targets(someone with a larger followback ratio), but you have to check if someone is in two different lists...

**Comment by tiranin** [comment]
And no I didnt have any issue, I do have issues with commenting, cause the limit is really short, and the bot mark as commented even If your comment is not accepted in instagram. I run headless, so I suspect some comments r not posted.
